### ISSUE TITLE
FPS DROP Only on my DarkRP Server and not Others

### ISSUE THREAD

**Issue by anyome** [Facepunch/garrysmod-issues#4509]
Hello ! @willox  & @robotboy655 

I have a 40-50 Players daily darkrp server, and a lot of my players say that their FPS Drop after (1 hour or so) of playing (Their fps goes from: 30 to 10).

I have the same configs as others DarkRP server, and this FPS Drop bug occurs only on my Server and not on : Havanna DarkRP. (151.80.111.13:27015)

It looks like they are using; Multi Thread server-side or using some specific sv_maxrate and sv_minrate configs to keep their FPS on their players

Did i forgot to add a launch Option to allow Multi Threading (ServerSide)?
Is there an option that We Server Owners can add on our Servers that i'm missing that gives players better FPS experience?

Thanks a lot !

**Comment by GitSparTV** [comment]
There's no such thing is "Multi Treaded ServerSide". FPS is a clientside value. Unless you are speaking about tickrate

**Comment by anyome** [comment]
I know, i wanna know what config does this server :"Havanna DarkRP" uses that makes their FPS on client stay high all the time?

I'm using the same tickrate as theirs (16 Tickrate)
And we are using the same addons.

I don't understand why they got bigger fps than us?

They are using even more addons than us!

**Comment by anyome** [comment]
The problem is the "FPS Drop" after a lot of play time on Gmod Server.

Basically, after 1 Hour of Playing on a Gmod Server, players need to "retry" and join again, because they get 10 FPS. and this bug occurs mostly on all Gmod Servers.

The FPS automatically DROP after 1 hour of playing, and all the players need to Retry or they'll be playing with 10 FPS.

Occurs basically on all the DarkRP Server but not on Havanna DarkRP

It's strange, they are using some "life saver" config that i really wanna know!

**Comment by GitSparTV** [comment]
Are you sure you have everything as they have? That would be copying lol.
The reason can be any.
Memory leak, overloaded render, server is not restarting at least once a day, etc

**Comment by anyome** [comment]
We have the greatest machine ever!

We are restarting our servers everyday, and we have an Intel® Core™ i7-4790 4GHz.

We are running only one darkrp Server and we have this Problem!


I'm sure they are using some configs that makes their server have great FPS!

This config should be on all Gmod Servers because that is considered as "Unfair" to others gmod servers.

They get more players because they have this bug Fixed, and all others darkrp server are dying.

It's a trick that only few people know about, and they don't wanna share it.

I really think that @robotboy655 should add it automatically after figuring what is it on the next May Update, and makes all server owners have it on their servers.

Because this is quite Unfair that somes Servers have a trick to have better FPS than others.

**Comment by GitSparTV** [comment]
With the same reason you can add an anticheat on all servers. That's unfair that we should find a good anticheat or buy it!

**Comment by Earu** [comment]
The only thing affecting client FPS are the addons you have installed on your server. Theres no magical config to have your clients gain more "fps". Also from the vague description you're making how do you expect anyone to know exactly what they're doing? You could start by providing us with an actual config file so we have an idea of what you're talking about in the first place.

**Comment by robotboy655** [comment]
What @Earu said basically. This topic does not belong here, and there's no magical config to fix your server. The problem is either with your addons or the gamemode itself. Debug it yourself. it takes only 1 addon to completely ruin performance. Amount of addons is meaningless, although the more you have, the higher the chance you will encounter a bad performing one.

**Comment by viral32111** [comment]
Oh it's this guy again... (#4465), I'm just going to summarise what I said before. Try using a performance profiling/debugging addon to identify clientside hooks and functions that could be diminishing FPS, then either remove them or find somebody who knows how to optimise said script/addon. Most of the time for FPS hitters, it'll be unoptimised code that runs over and over again in Think, Tick and HUD hooks.

A big FPS impact is usually maps themselves, since you're running DarkRP I'm going to assume you're using a fairly big sized map with lots of entities, buildings, areas, etc. If the isn't heavily optimised then players are going to be rendering/processing poritions of the map that're pointless to their current position.

Generally speaking, it doesn't matter if your server is using the top-tier preforming hardware so long as it can keep up with sending/receiving lots of networked data at a reasonable pace, because as you say it's clientside FPS related issues you're experiencing. Have you considered that maybe the clients playing on your servers are just using crap hardware on their end? Fixing clientside lag isn't simply a config option on the server, it's a number of factors including the actual client -_-

As for the "they start lagging after <x> amount of time", that sounds like a memory leak issue (like Spar said) that's causing excess memory to be used up on your clients and eventually their hitting the cap.

TL;DR: If you're planning on creating a big server, don't install addons randomly. Only install addons that you know are optimised. Hire developers (or learn yourself) to go through code and remove absolutely everything you aren't going to use. Rewrite code that's flagged in profiling addons. Disable parts of the gamemode you aren't using. Optimise your map.

(Since you've opened two issues related to performance and as far as everyone else can tell it's not related to the game, this should be asked/continued in the Discord or anywhere away from this issue tracker generally.)
